code/modules/mob/dead/observer/observer_base.dm

```diff
@@ -375,6 +375,74 @@ This is the proc mobs get to turn into a ghost. Forked from ghostize due to comp
 		return
 	GLOB.ghost_hud_panel.ui_interact(src)
 
+/mob/dead/observer/verb/respawn_character()
+	set name = "Respawn as New Character"
+	set category = "Ghost"
+
+	if(!GLOB.configuration.general.respawn_enabled && !check_rights(R_ADMIN))
+		to_chat(usr, "<span class ='warning'>Respawning has been disabled.</span>")
+		return
+
+	if(stat != DEAD)
+		to_chat(usr, "<span class ='notice'>You must be dead to respawn!</span>")
+		return
+
+	if(!SSticker || SSticker.current_state < GAME_STATE_PLAYING)
+		to_chat(src, "<span class ='warning'>You can't respawn before the game starts!</span>")
+		return
+
+	if(GAMEMODE_IS_WIZARD || GAMEMODE_IS_NUCLEAR)
+		to_chat(usr, "<span class ='warning'>You can't respawn for this gamemode.</span>")
+		return
+
+	var/deathtime = world.time - timeofdeath
+	if(!HAS_TRAIT(src, TRAIT_RESPAWNABLE) && !check_rights(R_ADMIN))
+		to_chat(usr, "<span class ='warning'>You don't have respawnability!</span>")
+		return
+
+	var/deathtimeminutes = round(deathtime / 600)
+	var/pluralcheck = "minutes"
+	if(deathtimeminutes == 0)
+		pluralcheck = ""
+	else if(deathtimeminutes == 1)
+		pluralcheck = "[deathtimeminutes] minute and"
+	else if(deathtimeminutes > 1)
+		pluralcheck = "[deathtimeminutes] minutes and"
+	var/deathtimeseconds = round((deathtime - deathtimeminutes * 600) / 10,1)
+
+	if(deathtimeminutes < GLOB.configuration.general.respawn_delay / 600 && !check_rights(R_ADMIN))
+		to_chat(usr, "<span class ='notice'>You have been dead for [pluralcheck][deathtimeseconds] seconds.</span>")
+		to_chat(usr, "<span class ='warning'>You must wait [GLOB.configuration.general.respawn_delay / 600] minutes before you can respawn.</span>")
+		return
+	if(isobserver(usr) && HAS_TRAIT(src, TRAIT_RESPAWNABLE))
```

Comment:
Run this `isobserver` check a lot earlier, and then typecast `usr` to the mobtype and run off of that, instead of checking `src`. I think that's the play in verbs?